generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/extensions/kClassExtensions.kt

```diff
@@ -88,8 +89,19 @@ internal fun KClass<*>.getSimpleName(isInputClass: Boolean = false): String {
         ?: this.simpleName
         ?: throw CouldNotGetNameOfKClassException(this)
 
+    fun Annotation.isInputOnlyAnnotation(): Boolean = when (this) {
+        is GraphQLValidObjectLocations -> this.locations.contains(GraphQLValidObjectLocations.Locations.INPUT_OBJECT)
```

Comment:
this check is not correct -> locations is an array so this check would be true even if type can be applied for both input and output (we want to apply the suffix if it is applicable only on the input types)

Reply:
Ah, of course. Nice catch, thanks 👍 
I added a test case for this and fixed it.